pd/src/consensus/worker.rs

```diff
@@ -247,164 +241,235 @@ impl Worker {
             .height
             .try_into()
             .expect("height should be nonnegative");
-        let epoch = pending_block.set_height(height);
+        let epoch = pending_block.set_height(
```

Comment:
However, removing the `epoch_duration` from the inside of `PendingBlock` means we have to supply it here.